internal/pkg/config/watcher/watcher.go

```diff
@@ -106,6 +106,11 @@ func (c *K8sConfigMapWatcher) Watch(ctx context.Context, notifyMe chan<- interfa
 	for {
 		select {
 		case e = <-ch:
+			if e.Object == nil {
+				glog.Errorf("got nil event, restart watcher")
+				watcher.Stop()
+				return nil
```

Comment:
we should return an informative error here, like `fmt.Errorf("watch event object is nil - HTTP connection likely closed")`. It would also be useful to comment that restarting the watcher happens in `cmd/main.go`, and is not performed by anything in `internal/pkg/config/watcher` package.

Additionally, could you drop comments here linking to https://github.com/kubernetes/client-go/issues/334 and explaining the bug this codepath intends to fix?

Reply:
But how to stop watcher in `cmd/main.go`. Add a func `StopWatch` in `internal/pkg/config/watcher`, so `cmd/main.go` can stop watcher?

Reply:
Ah, we should also update https://github.com/tumblr/k8s-sidecar-injector/blob/master/cmd/main.go#L112-L118 so it wont fatal, only log an error. Then, the `for{}` will just call Watch again for us, and everything will be 👌

Reply:
I throw error about this to `cmd/main.go`, I am not sure the watcher will be stopped when function `Watch()` exit.